internal/proxy/README.md

```diff
@@ -17,16 +17,42 @@ dependency to the proxy:
 1.  Ensure that the new dependency is covered under one of the `notice`,
     `permissive`, or `unencumbered` licences under the categories defined by
     [Google Open Source](https://opensource.google.com/docs/thirdparty/licenses/).
-2.  Add the dependency to the GCS bucket:
+2.  Gather the new set of dependencies and sync them to the GCS bucket.
 
 ```bash
-./internal/proxy/makeproxy.sh proxy
+# This approach was suggested by:
+# https://github.com/go-modules-by-example/index/tree/master/012_modvendor
+
+# Create a temporary directory where we'll create a module download cache.
+tgp=$(mktemp -d)
+
+# Run this command in the master branch.
+# It runs "go mod download" in every module that we have in our repo,
+# filling the cache with all of the module dependencies we need.
+GOPROXY=file://$GOPATH/pkg/mod/cache/download GOPATH=$tgp ./internal/proxy/makeproxy.sh
+
+# Run the above command again in your branch that's adding a new dependency,
+# to ensure the cache has any new dependencies.
+
+# Move the temporary cache to modvendor/.
+rm -rf modvendor
+cp -rp $GOPATH/pkg/mod/cache/download/ modvendor
+
+# Clean up the temporary cache.
+GOPATH=$tgp go clean -modcache
+rm -rf $tgp
+
+# Synchronize modvendor to the proxy.
 
 # -n: preview only
 # -r: recurse directories
 # -c: compare checksums not write times
-# -d: source directory; value should match what you passed to makeproxy.sh
-$ gsutil rsync -n -r -c -d proxy gs://go-cloud-modules
-...
-# If it looks good, repeat without the -n.
+gsutil rsync -n -r -c modvendor gs://go-cloud-modules
```

Comment:
With this new procedure in place, it seems like either `-d` should stay in, or any `.../list` files should be appended to. (I think the latter is a better approach, but may require some scripting.)

Reply:
SGTM.